week4/MealsharingNET/Services/InMemoryReservationRepository.cs

```diff
@@ -0,0 +1,45 @@
+using MealsharingNET.Models;
+
+namespace MealsharingNET;
+
+public class InMemoryReservationRepository : IReservationRepository
+{
+    private List<Reservation> Reservations { get; set; } = new List<Reservation>(){
+        new Reservation()
+        {
+            ID=1,
+            NumberOfGuests=10,
+            MealID=2,
+            CreatedDate= DateTime.Now,
+                       ContactName= "shan",
+            ContactEmail="shan@gmail.com"
+        },
+        new Reservation()
+        {
+            ID=2,
+            NumberOfGuests=100,
+            MealID=1,
+            CreatedDate= DateTime.Now,
+            ContactName= "haider",
+            ContactEmail="haider@gmail.com"
+        }
+    };
+
+    public IEnumerable<Reservation> ListReservations()
+    {
+        return Reservations;
+    }
+    public void Add(Reservation reservation)
+    {
+        Reservations.Add(reservation);
+    }
+    public IEnumerable<Reservation> GetReservation(int id)
+    {
+        return Reservations.SingleOrDefault(reservation => reservation.ID == id);
```

Comment:
If you use SingleOrDefault, you will get the only entry with that id, so you can return the expected type of Reservation instead of IEnumerable<Reservation>

Reply:
Sorry I didnt understand that part

Reply:
I mean, when you use "SingleOrDefault", you only get one result with a unique ID. Therefore, the result will not be a list, but only a single reservation. In this case, instead of IEnumerable of Reservation, you should use Reservation as output.
If it is not clear, tell me to explain in another way.

Reply:
I updated according to my understanding. Can you please look on update if its not correct then I think I didnt understand